kubernetes/resource_kubernetes_service_account.go

```diff
@@ -106,14 +106,18 @@ func resourceKubernetesServiceAccountCreate(d *schema.ResourceData, meta interfa
 		for _, secret := range secretList.Items {
 			for _, svcSecret := range diff {
 				if secret.Name != svcSecret.Name {
-					break
+					continue
```

Comment:
Here we want to continue to iterate over secrets to find the right one instead of breaking loop on first fail